src/container/Schedules/ScheduleContainer.tsx

```diff
@@ -135,68 +265,104 @@ export const ScheduleContainer = ({ tabs, data }: ScheduleContainerProps) => {
           tabs={tabs}
           handleAddClicked={() => setIsNewScheduleOpen(true)}
         />
-
         <ScheduleToolbar
-          handleNewServiceClicked={() => setIsNewServiceOpen(true)}
-          destroySchedule={() => setWarningDialog(SCHEDULE)} // this function should actually be moved into tabs. when a user right clicks the tab, you'd expect the delete functionality to be there**
+          handleNewServiceClicked={addService}
+          destroySchedule={() => setWarningDialog(SCHEDULE)}
           isScheduleModified={isScheduleModified}
           onSaveScheduleChanges={onSaveScheduleChanges}
         />
         <ContextMenu
           outerRef={outerRef}
           addRowHandler={insertRow}
-          deleteRowHandler={deleteRow}
+          deleteRowHandler={removeEvent}
         />
         <Prompt
           when={isScheduleModified}
           message="You have unsaved changes, are you sure you want to leave? Unsaved changes will be lost"
         />
         {!data.schedules && <div style={{ height: '50vh' }}></div>}
         {/* {alert && <Alert alert={alert} unMountAlert={() => setAlert(null)} />} */}
-        {data.schedules?.map((schedule: ScheduleTableInterface, idx) => {
-          const { columns: headers, services: bodies, title, view } = schedule;
+        {dataModel.schedules.map((schedule: ScheduleTableInterface, scheduleIndex) => {
```

Comment:
```suggestion
        {dataModel.schedules?.map((schedule: ScheduleTableInterface, scheduleIndex) => {
```
Otherwise I get a `dataModel.schedules is undefined` error.